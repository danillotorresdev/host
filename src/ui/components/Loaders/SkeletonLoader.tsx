import { styled } from "@linaria/react";

type SkeletonLoaderProps = {
  width?: number | string;
  height?: number | string;
  borderRadius?: number | string;
  backgroundColor?: string;
  className?: string;
};

const SkeletonLoaderElement = styled.div<SkeletonLoaderProps>`
  width: ${(props) => props.width ?? "100%"};
  height: ${(props) => props.height ?? "20px"};
  background-color: ${(props) => props.backgroundColor ?? "#f0f0f0"};
  border-radius: ${(props) => props.borderRadius ?? "4px"};
  margin-bottom: 10px;
`;

export const SkeletonLoader = ({
  width,
  height,
  borderRadius,
  backgroundColor,
  className,
}: SkeletonLoaderProps) => (
  <SkeletonLoaderElement
    className={className}
    width={width}
    height={height}
    borderRadius={borderRadius}
    backgroundColor={backgroundColor}
  />
);
