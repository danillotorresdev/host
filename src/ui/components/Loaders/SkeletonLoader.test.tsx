import { SkeletonLoader } from "./SkeletonLoader";
import { render } from "@/utils/settings/tests/utilities";
import { BrowserRouter } from "react-router-dom";
import { expect, it, describe } from "vitest";

describe("SkeletonLoader", () => {
  it("should render the user's name", () => {
    const { asFragment } = render(
      <SkeletonLoader />,
      {
        wrapper: BrowserRouter,
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});