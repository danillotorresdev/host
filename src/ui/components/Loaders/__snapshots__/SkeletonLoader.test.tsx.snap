// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SkeletonLoader > should render the user's name 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 s5rd3pb"
    style="--s5rd3pb-0: 100%; --s5rd3pb-1: 20px; --s5rd3pb-2: #f0f0f0; --s5rd3pb-3: 4px;"
  />
</DocumentFragment>
`;
